### PR TITLE
InfluxDB: Update flux library to fix support for boolean label values

### PR DESCRIPTION
**Pull request by ryantxu**
TimeSeries queries with boolean values can return errors.  This PR aims to fix that.

### CODE REVIEW

File: pkg/tsdb/influxdb/flux/executor_test.go [hunk 1]
```diff
@@ -204,6 +204,12 @@ func TestBuckets(t *testing.T) {
 	})
 }
 
+func TestBooleanGrouping(t *testing.T) {
+	t.Run("Boolean", func(t *testing.T) {
```
Comment (aknuds1) [inline pkg/tsdb/influxdb/flux/executor_test.go:208]: Why do you need a sub-test? Doesn't seem to add anything from the base test case.

**Review by aknuds1 — APPROVED** [review]
Generally LGTM.

File: pkg/tsdb/influxdb/flux/executor_test.go (hunk 1 above)
Comment (ryantxu) [reply pkg/tsdb/influxdb/flux/executor_test.go:208]: The test setup could maybe use some other style, but in general, this checks that:
[boolean.csv](https://github.com/grafana/grafana/pull/29310/files#diff-6fa002d068edadc27b2d0c767d5af644f57ddfb7b69c70e332552d8eb7c5bbc8)

becomes:
[boolean.golden.txt](https://github.com/grafana/grafana/pull/29310/files#diff-82729d376d3264b309c15f3ffea879cedf042dd39aff2c4b3271ab6941537a78)

Comment (aknuds1) [reply pkg/tsdb/influxdb/flux/executor_test.go:208]: Yeah I think the style of the test is fine, I just don't see why it needs to be wrapped in a sub-test :)

Comment (ryantxu) [reply pkg/tsdb/influxdb/flux/executor_test.go]: ahh -- cool.. I just remove all the redundant `t.Run` calls -- so now it is a big PR on the tests, but no real changes

**Comment by torkelo** [comment]
Anything we want to include in a patch release?

**Comment by ryantxu** [comment]
yes -- this could go in the patch release if possible

**Comment by torkelo** [comment]
Should this close #28976 ?

**Comment by torkelo** [comment]
was a conflict so did a manual backport https://github.com/grafana/grafana/pull/29333

**Pull request merged**